Sources/PubNub/EventEngine/Subscribe/Effects/EmitMessagesEffect.swift

```diff
@@ -37,14 +37,14 @@ class MessageCache {
 struct EmitMessagesEffect: EffectHandler {
   let messages: [SubscribeMessagePayload]
   let cursor: SubscribeCursor
-  let listeners: [BaseSubscriptionListener]
+  let listeners: WeakSet<BaseSubscriptionListener>
```

Comment:
what happens if someone adds a listener to a subscription, but doesn't keep a reference to it otherwise?

for example, if I did this in Kotlin with weak references:
```
subscription.addListener ( object : EventListener { override fun onMessage() {...} } )
```

then after a while, I would stop receiving events because no one holds a strong reference to the listener, so it gets garbage collected

Reply:
It looks like those names are extremely confusing. In fact, the `listeners` variable keeps instances to all subscriptions. This could be either `Subscription`, `SubscriptionSet`, or the item like `SubscriptionListener` which comes from the "old" API that we still have to support. So perhaps the best idea would be replacing variable's name from the line 40 with `subscriptions` or something less confusing.

Regarding your code snippet, classes like `Subscription` and `SubscriptionSet` internally store their listeners as **strong references**. Thus, you'll be able to receive events